File: chain-sdk/clojure/src/on_chain_api/specs/token_hour_data_v3_dto.clj
(ns on-chain-api.specs.token-hour-data-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def token-hour-data-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :period_start_unix) int?
   (ds/opt :token) string?
   (ds/opt :volume) string?
   (ds/opt :volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :total_value_locked) string?
   (ds/opt :total_value_locked_usd) string?
   (ds/opt :price_usd) string?
   (ds/opt :fees_usd) string?
   (ds/opt :open) string?
   (ds/opt :high) string?
   (ds/opt :low) string?
   (ds/opt :close) string?
   (ds/opt :vid) int?
   })

(def token-hour-data-v3-dto-spec
  (ds/spec
    {:name ::token-hour-data-v3-dto
     :spec token-hour-data-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/TokenHourDataV3DTO.js
goog.provide('API.Client.TokenHourDataV3DTO');

/**
 * Token data aggregated across all pairs that include token.
 * @record
 */
API.Client.TokenHourDataV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.recvTime;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.blockNumber;

/**
 * token address concatendated with date
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.id;

/**
 * unix timestamp for start of hour
 * @type {!number}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.periodStartUnix;

/**
 * pointer to token
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.token;

/**
 * volume in token units
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.volume;

/**
 * volume in derived USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.volumeUsd;

/**
 * volume in USD even on pools with less reliable USD values
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.untrackedVolumeUsd;

/**
 * liquidity across all pools in token units
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.totalValueLocked;

/**
 * liquidity across all pools in derived USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.totalValueLockedUsd;

/**
 * price at end of period in USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.priceUsd;

/**
 * fees in USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.feesUsd;

/**
 * opening price USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.open;

/**
 * high price USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.high;

/**
 * low price USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.low;

/**
 * close price USD
 * @type {!string}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.close;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenHourDataV3DTO.prototype.vid;

